core/src/main/scala/org/apache/spark/serializer/SerializationDebugger.scala

```diff
@@ -125,6 +125,13 @@ private[spark] object SerializationDebugger extends Logging {
       return List.empty
     }
 
+    /**
+     * Visit an externalizable object.
+     * Since writeExternal() can choose add arbitrary objects at the time of serialization,
+     * the only way to capture all the objects it will serialize is by using a
+     * dummy ObjectOutput object that captures all the inner objects, and then visit all the
```

Comment:
This is a really long sentence and fairly low level. How about:
```
Since writeExternal() can choose add arbitrary objects at the time of serialization,
the only way to capture all the objects it will serialize is to manually write the object
into our own dummy ObjectOutput that collects the relevant fields.
```